### ISSUE TITLE
Create Prefect flow tasks to create and destroy test API

### ISSUE THREAD

**Issue by mddilley** [cityofaustin/atd-data-tech#9703]
To create a test API for each test instance, we should create and destroy test Lambda and API Gateway resources that are currently deployed using [Zappa](https://github.com/zappa/Zappa) in moped-test, staging, and production using a secret config.

Another layer of this is creating a configuration in Secrets Manager so that the API can get the corresponding env vars from Secrets Manager. This is the way things work in the current test, staging, and production.

**Steps so far**
- [x] Create task to deploy
- [x] Create task to delete
- [x] Create task to create API configuration in Secrets Manager
- [x] Create task to remove API configuration from Secrets Manager
- [x] Make sure API is getting all of the secrets that it needs

**Comment by mddilley** [comment]
Update: Deploying/undeploying the API is working now.

After running into some blockers:
- AWS/zappa didn't like Python 3.10 (Frank helped me roll back to an earlier ubuntu image and update requirements)
- API needs a secret name passed to it to get from Secrets Manager (passing the same naming conventions used for the secret created in the secrets create/remove tasks)
- Make sure API can operate without errors (working on this)

**Comment by mddilley** [comment]
I've pulled in code updates from [Frank's ECS branch](https://github.com/cityofaustin/atd-moped/tree/refine-ecs-deployment) so that we align our flows and env vars. 

Also, I did some troubleshooting on why the API's health check was failing. Turns out that the zappa deploy command did not like having a partial ARN passed in its permissions config for the API config secret in Secrets Manager.

I updated the deploy/undeploy flows to depend on getting that ARN from the secret create/destroy tasks, and API was responsive after that.
